### PR TITLE
Hds starterkit version

### PR DESCRIPTION
**Pull request by Tjocksmurf**

### CODE REVIEW

**Comment by Tjocksmurf** [comment]
[Update Health Design System to latest version of Gold Design System](https://trello.com/c/EOTtbaGi/368-update-health-design-system-to-latest-version-of-gold-design-system)

File: bin/setup.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#! /usr/bin/env node
+const path = require('path');
+const fs = require('fs-extra');
+var getDeps = require('get-dependencies');
+const {getInstalledPathSync} = require('get-installed-path');
+
+console.log("Creating Health Design System starter kit base.");
+
+//Collect paths
```
Comment (Benjen) [inline bin/setup.js]: Could you make sure that file comments are in the correct format.

File: bin/setup.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+const gold = getInstalledPathSync('@gold.au', {local: true});
+const hds = getInstalledPathSync('@health.gov.au/health-design-system', {local: true});
+const base_path = process.cwd();
+const srcDir = hds + '/source';
+const destDir = process.cwd() + `/assets/hds`;
+
+
+//Copy files to assets folder
```
Comment (Benjen) [inline bin/setup.js]: Could you make sure that file comments are in the correct format.

File: bin/setup.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+try {
+  fs.copySync(srcDir, destDir, {overwrite: true})
+  console.log("■ Files copied to assets folder");
+} catch (err) {
+  console.error(err)
+}
+
+//Check if assets/scss/main.scss exists, if not create it with reference.
```
Comment (Benjen) [inline bin/setup.js]: Could you make sure that file comments are in the correct format.

File: bin/setup.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+} else {
+  fs.writeFile(base_path + '/assets/scss/main.scss', "@import '../hds/sass/all.scss';", function (err) {
+    if (err) {
+      return console.log(err);
+    }
+    console.log("■ Created /assets/scss/main.scss and added import statement for HDS");
+  });
+}
+
+//Add GOLD dependencies
```
Comment (Benjen) [inline bin/setup.js]: Could you make sure that file comments are in the correct format.

File: bin/setup.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+    r.unshift('\n//GOLD SCSS dependencies. ');
+
+    fs.appendFile(process.cwd() + `/assets/hds/sass/all.scss`, r.join("\n"), function (err) {
+      if (err) throw err;
+      console.log('■ GOLD references written to scss file.');
+    });
+  });
+
+//Copy and rename pancake ref file
```
Comment (Benjen) [inline bin/setup.js]: Could you make sure that file comments are in the correct format.

File: bin/setup.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  fs.copyFile(base_path + '/pancake/sass/pancake.scss', base_path + '/assets/hds/sass/vendors/govau-ds.scss', (err) => {
+    if (err) throw err;
+    console.log('■ Pancake reference file copied and renamed.');
+  });
+} else {
+  console.log('\x1b[31m%s\x1b[0m','■ WARNING! No pancace folder found, cannot copy file. Delete the node_modules folder and try again.');
+}
+
+
+//Remove pancake folder
```
Comment (Benjen) [inline bin/setup.js]: Could you make sure that file comments are in the correct format.

File: bin/setup.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      return console.log(err);
+    }
+    console.log("■ Created /assets/scss/main.scss and added import statement for HDS");
+  });
+}
+
+//Add GOLD dependencies
+getDeps.getByFile(hds + '/package.json')
+  .then(function (result) {
+    let r = result.filter(val => val.startsWith('@gold') && !val.includes('pancake')).map(filt => '@import "' + base_path + '/node_modules/' + filt + '/lib/sass/_module.scss";');
```
Comment (Benjen) [inline bin/setup.js]: [optional] this is a little on the long side with too many layers. Makes it quite difficult to read. Can't really follow what is going on.

**Review by Benjen — CHANGES_REQUESTED** [review]

File: bin/setup.js (hunk 1 above)
Comment (Tjocksmurf) [reply bin/setup.js]: I've added eslint to the project and applied airbnb js standards.

File: bin/setup.js (hunk 7 above)
Comment (Tjocksmurf) [reply bin/setup.js]: I have added comments to explain what is going on.

**Review by Benjen — CHANGES_REQUESTED** [review]
Suggested changes:
- Add documentation about the `npx hdssetup` command.

**Pull request merged**